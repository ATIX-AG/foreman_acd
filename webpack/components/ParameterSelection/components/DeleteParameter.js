import React from 'react';
import PropTypes from 'prop-types';
import {
  Icon,
  Button,
} from 'patternfly-react';

const DeleteParameter = ({
  hidden,
  disabled,
  onDeleteParameter,
  additionalData,
}) =>{
  if (hidden) {
    return null;
  }

  return (
    <span>
      &nbsp;
      <Button
        bsStyle="default"
        disabled={disabled}
        onClick={() => window.confirm("Are you sure you wish to delete this item?") && onDeleteParameter(additionalData) }
      >
        <Icon type="pf" name="delete" />
      </Button>
    </span>
  );
};

DeleteParameter.propTypes = {
  hidden: PropTypes.bool.isRequired,
  disabled: PropTypes.bool.isRequired,
  onDeleteParameter: PropTypes.func.isRequired,
  additionalData: PropTypes.object.isRequired,
};

export default DeleteParameter;
