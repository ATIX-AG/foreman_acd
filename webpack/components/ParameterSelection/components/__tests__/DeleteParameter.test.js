import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import DeleteParameter from '../DeleteParameter';

const noop = () => {};

const fixtures = {
  'should render delete parameter': {
    hidden: false,
    disabled: false,
    onDeleteParameter: noop,
    additionalData: {},
  },
  'should render hidden button': {
    hidden: true,
    disabled: false,
    onDeleteParameter: noop,
    additionalData: {},
  },
  'should render disabled button': {
    hidden: false,
    disabled: true,
    onDeleteParameter: noop,
    additionalData: {},
  },
};

describe('DeleteParameter', () =>
  testComponentSnapshotsWithFixtures(DeleteParameter, fixtures));
