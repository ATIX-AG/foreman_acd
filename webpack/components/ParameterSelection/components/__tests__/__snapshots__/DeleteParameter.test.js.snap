// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteParameter should render delete parameter 1`] = `
<span>
   
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}
  >
    <Icon
      name="delete"
      type="pf"
    />
  </Button>
</span>
`;

exports[`DeleteParameter should render disabled button 1`] = `
<span>
   
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={true}
    onClick={[Function]}
  >
    <Icon
      name="delete"
      type="pf"
    />
  </Button>
</span>
`;

exports[`DeleteParameter should render hidden button 1`] = `""`;
